src/java/org/apache/cassandra/index/sai/disk/StorageAttachedIndexWriter.java

```diff
@@ -126,9 +127,14 @@ public void nextUnfilteredCluster(Unfiltered unfiltered)
         if (!unfiltered.isRow())
             return;
 
+        // Ignore rows with no live data...
+        Row row = (Row) unfiltered;
+        if (!row.hasLiveData(FBUtilities.nowInSeconds(), false))
+            return;
```

Comment:
The analog of what's been done in the `Indexer` above (but for compaction and initial index building).